File: src/kt2/ch8/Lec08Main.java
package kt2.ch8;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class Lec08Main {

    public static void main(String[] args) throws IOException {

        Lec08Main lec07Main = new Lec08Main();

    }


    public int max(int a, int b) {
        if (a > b) {
            return a;
        }
        return b;
    }

    public void repeat(String str, int num, boolean useNewLine) {
        for (int i = 1; i <= num; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.println(str);
            }
        }
    }

    public void repeat(String str, int num) {
        repeat(str, num, true);
    }

    public void repeat(String str) {
        repeat(str, 3, true);
    }

    public void printAll(String... strings){
        for (String string : strings) {
            System.out.println(string);
        }
    }

}

File: src/kt2/ch8/Lec08Main.kt
package kt2.ch8

/**
 * public은 생략가능, 기본이 public
 *
 * 함수가 하나의 결과값이면 block 대신 = 사용가능
 * = 을 사용하는 경우 반환타입 생략가능
 *
 * default paramter 지원, named argument 지원
 *
 * 가변인자
 *
 * */

fun main() {

    repeat("Hello World", useNewLine = false)

    val array = arrayOf("A", "B", "C")
    printAll(*array)

}

fun max(a: Int, b: Int) = if (a > b) a else b


fun repeat(str: String?,
           num : Int = 3,
           useNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (useNewLine) {
            println(str)
        } else {
            println(str)
        }
    }
}

fun printAll(vararg strings: String?) {
    for (string in strings) {
        println(string)
    }
}


